docs/pages/router/reference/api-routes.mdx

```diff
@@ -0,0 +1,287 @@
+
+export async function GET(request: ExpoRequest, { post }: Record<string, string>) {
+  if (!post) {
+    return new ExpoResponse('No post found', {
+      status: 404,
+      headers: {
+        'Content-Type': 'text/plain',
+      },
+    });
+  }
+  // fetch data for `post`
+  return ExpoResponse.json({ ... });
+}
+```
+
+Making requests with an undefined method will automatically return `405: Method not allowed`.
+
+If an error is thrown during the request, it will automatically return `500: Internal server error`.
+
+## Bundling
+
+API Routes are bundled with Expo CLI and [Metro bundler](/guides/customizing-metro), meaning they have access to all of the same language features as your client code.
+
+- [TypeScript](/guides/typescript) –– types and [`tsconfig.json` paths](/guides/typescript#path-aliases).
+- [Environment variables](/guides/environment-variables) –– server routes have access to all environment variables, not just the ones prefixed with `EXPO_PUBLIC_`.
+- Node.js standard library –– ensure that you are using the correct version of Node.js locally for your server environment.
+- `babel.config.js` and `metro.config.js` support –– settings work across both client and server code.
+
+{/* TODO: SSR, redirects, middleware, etc. */}
+
+## Security
+
+> While in beta, server code may leak into the client bundle. This won't happen in the stable release.
+
+Route handlers are executed in a sandboxed environment that is isolated from the client code. This means that you can safely store sensitive data in the route handlers without exposing it to the client.
+
+- If your client code imports code with a secret, it will still be included in the client bundle. This also applies to ALL files in the `app` directory that are not a route handler (suffixed with `+api.js`).
+- If your secret is in a `<...>+api.js` (Route Handler) file, it will not be included in the client bundle. This applies to all files that are imported in the route handler.
+- The secret stripping takes place in `expo/metro-config` and requires this be used in your `metro.config.js`!
+
+## Deployment
+
+> This is experimental and subject to breaking changes. We have no continuous tests against this configuration.
+
+Every cloud hosting provider needs a custom adapter to support the Expo server runtime. The following providers have unofficial/experimental support from the Expo team.
+
+Before you deploy with third-parties, it's good to be familiar with the basics of the [`npx expo export`](/more/expo-cli#exporting) command. Specifically:
+
+- `/dist` is the default export directory for Expo CLI.
+- Files in `/public` will be copied to `/dist` on export.
+- The `@expo/server` package is included with `expo` and delegates requests to the server routes.
+- `@expo/server` does **not** inflate environment variables from `.env` files, these are expected to be loaded either by the hosting provider or the user.
+- Metro is not included in the server.
+
+### Express
```

Comment:
Did you test with Bun yet? If not, I'd be happy to write the Bun equivalent as we have for Express.

Reply:
We should wait until this lands before adding any additional adapters.